rct229/rulesets/ashrae9012019/ruleset_functions/compare_swh_dist_systems_and_components_test.py

```diff
@@ -0,0 +1,143 @@
+                                {
+                                    "id": "Space 3",
+                                },
+                            ],
+                        }
+                    ],
+                }
+            ],
+        }
+    ],
+    "service_water_heating_distribution_systems": [
+        {
+            "id": "SWH Distribution 1",
+            "tanks": [
+                {
+                    "id": "Tank 1",
+                },
+                {
+                    "id": "Tank 2",
+                },
+            ],
+            "service_water_piping": [
+                {
+                    "id": "SWH Piping 1",
+                    "child": [
+                        {
+                            "id": "SWH Piping Child 1",
+                            "child": [
+                                {
+                                    "id": "SWH Piping 1-a",
+                                },
+                                {
+                                    "id": "SWH Piping 1-b",
+                                },
+                            ],
+                        }
+                    ],
+                },
+                {
+                    "id": "SWH Piping 2",
+                },
+            ],
+        }
+    ],
+    "pumps": [
+        {
+            "id": "Pump 1",
+            "loop_or_piping": "SWH Piping 1",
+        },
+        {
+            "id": "Pump 2",
+            "loop_or_piping": "SWH Piping 2",
+        },
+    ],
+    "service_water_heating_equipment": [
+        {
+            "id": "SWH Equipment 1",
+            "distribution_system": "SWH Distribution 1",
+            "solar_thermal_systems": [
+                {
+                    "id": "Solar Thermal System 1",
+                },
+                {
+                    "id": "Solar Thermal System 2",
+                },
+            ],
+        },
+        {
+            "id": "SWH Equipment 2",
+            "distribution_system": "SWH Distribution 2",
+            "solar_thermal_systems": [
+                {
+                    "id": "Solar Thermal System 3",
+                },
+                {
+                    "id": "Solar Thermal System 4",
+                },
+            ],
+        },
+    ],
+    "type": "BASELINE_0",
+}
+
+TEST_RPD_FULL = {
+    "id": "229",
+    "ruleset_model_descriptions": [TEST_RMD],
+    "data_timestamp": "2024-02-12T09:00Z",
+}
+
+TEST_RMD = quantify_rmd(TEST_RPD_FULL)["ruleset_model_descriptions"][0]
+
+
+def test__TEST_RPD_FIXED_TYPE__is_valid():
+    schema_validation_result = schema_validate_rmd(TEST_RPD_FULL)
+    assert schema_validation_result[
+        "passed"
+    ], f"Schema error: {schema_validation_result['error']}"
+
+
+def test__get_aggregated_zone_hvac_fan_operating_schedule__correct_mapping():
+    assert (
+        compare_swh_dist_systems_and_components(
+            TEST_RMD, TEST_RMD, "test", "SWH Distribution 1"
```

Comment:
I wasn't sure what the ```compare_context_str``` means, so for now I input ```test``` value for now, but I'll correct this when I have better an idea of this argument.

Reply:
And, I'm going to add more unit tests as well